### PR TITLE
Post body changes for `validate_vms_resource ` to include service template_id

### PR DESCRIPTION
**Pull request by AparnaKarve**
Dependent on https://github.com/ManageIQ/manageiq/pull/18065
 
Fixes ManageIQ/manageiq-v2v#677

(Tests are expected to fail until https://github.com/ManageIQ/manageiq/pull/18065 is merged)

### CODE REVIEW

**Comment by AparnaKarve** [comment]
@miq-bot add_label hammer/yes

**Comment by AparnaKarve** [comment]
@bdunne Can you review this?

ManageIQ/manageiq#18065 was just merged after which I restarted Travis here.

File: spec/requests/transformation_mappings_spec.rb [hunk 1]
```diff
@@ -305,6 +305,79 @@ def href_slug(obj)
...
+
+        context "can validate vms with csv data and service_template_id are specified" do
+          it "vm belongs to the service_template record" do
+            api_basic_authorize(action_identifier(:transformation_mappings, :validate_vms, :resource_actions, :post))
+            source_ems = FactoryGirl.create(:ems_cluster)
+            destination_ems = FactoryGirl.create(:ems_cluster)
+            transformation_mapping =
+              FactoryGirl.create(:transformation_mapping,
+                                 :transformation_mapping_items => [TransformationMappingItem.new(:source => source_ems, :destination => destination_ems)])
+            vm = FactoryGirl.create(:vm_vmware, :name => 'test_vm', :ems_cluster => source_ems, :ext_management_system => FactoryGirl.create(:ext_management_system))
```
Comment (bdunne) [inline spec/requests/transformation_mappings_spec.rb]: Please don't set `name` as there's a sequencer for it.

File: spec/requests/transformation_mappings_spec.rb [hunk 2]
```diff
@@ -305,6 +305,79 @@ def href_slug(obj)
...
+          end
+
+          it "vm does not belong to the service_template record" do
+            api_basic_authorize(action_identifier(:transformation_mappings, :validate_vms, :resource_actions, :post))
+            source_ems = FactoryGirl.create(:ems_cluster)
+            destination_ems = FactoryGirl.create(:ems_cluster)
+            transformation_mapping =
+              FactoryGirl.create(:transformation_mapping,
+                                 :transformation_mapping_items => [TransformationMappingItem.new(:source => source_ems, :destination => destination_ems)])
+            vm = FactoryGirl.create(:vm_vmware, :name => 'test_vm', :ems_cluster => source_ems, :ext_management_system => FactoryGirl.create(:ext_management_system))
```
Comment (bdunne) [inline spec/requests/transformation_mappings_spec.rb]: Same here

File: spec/requests/transformation_mappings_spec.rb [hunk 3]
```diff
@@ -305,6 +305,79 @@ def href_slug(obj)
           expect(updated_mapping_destination).to match_array(reference_destination)
           expect(transformation_mapping.transformation_mapping_items.count).to eq(TransformationMappingItem.all.count)
         end
+
+        context "can validate vms with csv data and service_template_id are specified" do
+          it "vm belongs to the service_template record" do
+            api_basic_authorize(action_identifier(:transformation_mappings, :validate_vms, :resource_actions, :post))
+            source_ems = FactoryGirl.create(:ems_cluster)
```
Comment (bdunne) [inline spec/requests/transformation_mappings_spec.rb]: Is this supposed to be an EMS or a cluster?  Please adjust the factory or variable name accordingly.

File: spec/requests/transformation_mappings_spec.rb (hunk 1 above)
Comment (bdunne) [reply spec/requests/transformation_mappings_spec.rb]: Should one of the clusters above (assuming it's supposed to be a cluster) be connected to this `ext_management_system`?

File: spec/requests/transformation_mappings_spec.rb [hunk 4]
```diff
@@ -305,6 +305,79 @@ def href_slug(obj)
...
+              "invalid"    => [],
+              "conflicted" => []
+            }
+            expect(response).to have_http_status(:ok)
+            expect(response.parsed_body).to include(expected)
+          end
+
+          it "vm does not belong to the service_template record" do
+            api_basic_authorize(action_identifier(:transformation_mappings, :validate_vms, :resource_actions, :post))
+            source_ems = FactoryGirl.create(:ems_cluster)
```
Comment (bdunne) [inline spec/requests/transformation_mappings_spec.rb]: Same here

**Comment by AparnaKarve** [comment]
@bdunne Addressed your feedback for improving readability in the tests.
Can you review again?

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 91c220f5af72a6efb6a786dadda04c030a84ad60
Author: Brandon Dunne <brandondunne@hotmail.com>
Date:   Wed Oct 24 17:50:05 2018 -0400

    Merge pull request #486 from AparnaKarve/post_body_changes_to_include_service_template_id
    
    Post body changes for `validate_vms_resource ` to include service template_id
    
    (cherry picked from commit 93b07cdc03cfb951c5239e3850a011fc8c13b661)
```

**Pull request merged**